tensilelite/Tensile/SolutionStructs.py

```diff
@@ -2819,10 +2843,17 @@ def calcLdsNumBytes(ldsPadA: int, LdsBlockSizePerPadA: int, ldsPadB: int, LdsBlo
           if state["TransposeLDS"] and (not state["DirectToLds"]):
             state["LocalReadVectorWidth"] = 16 // state["ProblemType"]["DataType"].numBytes()
           else:
-            state["LocalReadVectorWidth"] = state["MIInputPerThread"]
+            if state["ProblemType"]["Sparse"] and state["MIInputPerThread"] * state["ProblemType"]["DataType"].numBytes() > 16:
+              state["LocalReadVectorWidth"] = 16 // state["ProblemType"]["DataType"].numBytes()
+            else:
+              state["LocalReadVectorWidth"] = state["MIInputPerThread"]
         else:
-          if state["LocalReadVectorWidth"] < state["MIInputPerThread"]:
-            reject(state, "LocalReadVectorWidth < %u" %(state["MIInputPerThread"]))
+          if state["ProblemType"]["Sparse"] and state["MIInputPerThread"] * state["ProblemType"]["DataType"].numBytes() > 16:
+            if state["LocalReadVectorWidth"] < state["MIInputPerThread"] // 2:
+              reject(state, "LocalReadVectorWidth < %u" %(state["MIInputPerThread"] // 2))
+          elif not globalParameters["AsmCaps"][globalParameters["CurrentISA"]]["HasMFMA_f8f6f4"]:
```

Comment:
should not use globalParameters["CurrentISA"],  please use state["isa"] or isa instead.